src/Listener/ApiPaginationListener.php

```diff
@@ -87,27 +87,33 @@ public function beforeRender(Event $event)
      */
     protected function _getJsonApiPaginationResponse(array $pagination)
     {
-        $self = Router::url([
+        if ($this->_controller()->Crud->config('listeners.jsonApi.absoluteLinks') === true) {
+            $routerMethod = 'url';
+        } else {
```

Comment:
The variables default state could be assigned regardless, and only if the condition matches would it be overwritten, which would remove the `else`. Just a thought 😄